staging/src/k8s.io/apiserver/pkg/server/egressselector/config.go

```diff
@@ -134,13 +131,13 @@ func validateDirectConnection(connection apiserver.Connection, fldPath *field.Pa
 
 func validateUDSConnection(connection apiserver.Connection, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if connection.TCP != nil {
+	if connection.Transport.TCP != nil {
```

Comment:
Since you have already checked the TCP config before calling validateUDSConnection, do you still need this check here?